pkg/build/controller/build/build_controller.go

```diff
@@ -1039,14 +1132,23 @@ func (bc *BuildController) podDeleted(obj interface{}) {
 // is created
 func (bc *BuildController) buildAdded(obj interface{}) {
 	build := obj.(*buildapi.Build)
+	// initiate gauge metric to new before any pod activity ... pod errors will transition from new to error state
+	metrics.UpdateActiveBuildsGauge("", string(buildapi.BuildPhaseNew), build.ObjectMeta.Namespace)
 	bc.enqueueBuild(build)
 }
 
 // buildUpdated is called by the build informer event handler whenever a build
 // is updated or there is a relist of builds
 func (bc *BuildController) buildUpdated(old, cur interface{}) {
-	build := cur.(*buildapi.Build)
-	bc.enqueueBuild(build)
+	oldbuild := old.(*buildapi.Build)
+	newbuild := cur.(*buildapi.Build)
+	if buildutil.IsTerminalPhase(newbuild.Status.Phase) {
+		metrics.UpdateActiveBuildsGauge(string(oldbuild.Status.Phase), "", newbuild.ObjectMeta.Namespace)
+		updateMetricsForTerminalBuild(newbuild)
+	} else {
+		metrics.UpdateActiveBuildsGauge(string(oldbuild.Status.Phase), string(newbuild.Status.Phase), newbuild.ObjectMeta.Namespace)
```

Comment:
ok I understand the duplicate updates now, I would have expected this logic to be in the transitionToPhase function instead, so it was only called when we were actually updating a build's phase from one state to another.  There could still be duplication scenarios even then, as we discussed, but they'd be much rarer.